src/OutputReaders/field_time_series_update.jl

```diff
@@ -0,0 +1,102 @@
+import Oceananigans.BoundaryConditions: BoundaryCondition, getbc
+import Oceananigans.Models: update_time_series!
+
+using Oceananigans.Fields: AbstractField
+using Oceananigans.AbstractOperations: AbstractOperation
+using Oceananigans.TimeSteppers: AbstractTimeStepper
+using Oceananigans.Models: AbstractModel
+
+using Oceananigans.TimeSteppers: Clock
+
+const CPUFTSBC = BoundaryCondition{<:Any, <:FieldTimeSeries}
+const GPUFTSBC = BoundaryCondition{<:Any, <:GPUAdaptedFieldTimeSeries}
+
+const FTSBC = Union{CPUFTSBC, GPUFTSBC}
+
+@inline getbc(bc::FTSBC, i::Int, j::Int, grid::AbstractGrid, clock::Clock, args...) = bc.condition[i, j, Time(clock.time)]
+
+# Set a field with a range of time indices.
+# We change the index range of the `FieldTimeSeries`
+# and load the new data
+function set!(fts::InMemoryFieldTimeSeries, index_range::UnitRange)
+    if fts.backend.index_range == 1:length(fts.times)
+        return nothing
+    end
+
+    fts.data.index_range .= index_range
+    set!(fts, fts.backend.path, fts.backend.name)
+
+    return nothing
+end
+
+# Update the `fts` to contain the time `time_index.time`.
+function update_time_series!(fts::InMemoryFieldTimeSeries, time_index::Time)
```

Comment:
the file is "update_field_time_series" but this function is "update_time_series". Let's use the same name throughout